### ISSUE TITLE
Grails error on viewing announcements on dev

### ISSUE THREAD

**Issue by ostephens** [k-int/KBPlus#680]
e.g. http://dev.kbplus.ac.uk/kbplus/myInstitutions/JISC_Collections/announcements
gives Grails error

Error 500: Internal Server Error
URI
/kbplus/myInstitutions/JISC_Collections/announcements
Class
java.lang.NullPointerException
Message
Cannot get property 'id' on null object
Trace

Line | Method
->>   56 | doFilter  in /myInstitutions/announcements.gsp

---

Caused by NullPointerException: Cannot get property 'id' on null object
->>   48 | doCall    in gsp_demo_myInstitutionsannouncements_gsp$_run_closure2

---

|     61 | run       in gsp_demo_myInstitutionsannouncements_gsp
|    195 | doFilter  in grails.plugin.cache.web.filter.PageFragmentCachingFilter
|     63 | doFilter  in grails.plugin.cache.web.filter.AbstractFilter
|     65 | doFilter  in com.k_int.kbplus.filter.ApiauthFilter
|   1110 | runWorker in java.util.concurrent.ThreadPoolExecutor
|    603 | run . . . in java.util.concurrent.ThreadPoolExecutor$Worker
^    722 | run       in java.lang.Thread

**Comment by ianibbo** [comment]
Looks like there is an announcement without a user - which shouldn't happen
:/ Have added some null checks. Will need to wait for a rebuild to
investigate further. Notice that there are records with diacritic problems
in the announcements page.

e

Ian Ibbotson
Director
Knowledge Integration Ltd
35 Paradise Street, Sheffield. S3 8PZ
T: 0114 273 8271
M: 07968 794 630
W: http://www.k-int.com

**Comment by ianibo** [comment]
OK.. Null safety check on dev now renders this page, but none of the
announcements have users attached to them. Bit confused since the test DB
does seem to have users attached - thought the DB was copied, so i would
expect it to seem the same on both test and dev..

Any ideas?

On 20 January 2014 14:59, Owen Stephens notifications@github.com wrote:

> e.g.
> http://dev.kbplus.ac.uk/kbplus/myInstitutions/JISC_Collections/announcements
> gives Grails error
> 
> Error 500: Internal Server Error
> URI
> /kbplus/myInstitutions/JISC_Collections/announcements
> Class
> java.lang.NullPointerException
> Message
> Cannot get property 'id' on null object
> Trace
> 
> Line | Method
> 
> ## ->> 56 | doFilter in /myInstitutions/announcements.gsp
> 
> Caused by NullPointerException: Cannot get property 'id' on null object
> 
> ## ->> 48 | doCall in gsp_demo_myInstitutionsannouncements_gsp$_run_closure2
> 
> | 61 | run in gsp_demo_myInstitutionsannouncements_gsp
> | 195 | doFilter in
> grails.plugin.cache.web.filter.PageFragmentCachingFilter
> | 63 | doFilter in grails.plugin.cache.web.filter.AbstractFilter
> | 65 | doFilter in com.k_int.kbplus.filter.ApiauthFilter
> | 1110 | runWorker in java.util.concurrent.ThreadPoolExecutor
> | 603 | run . . . in java.util.concurrent.ThreadPoolExecutor$Worker
> ^ 722 | run in java.lang.Thread
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/k-int/KBPlus/issues/680
> .

## 

Ian Ibbotson
W: http://ianibbo.me
E: ianibbo@gmail.com
skype: ianibbo
twitter: ianibbo

**Comment by ostephens** [comment]
We used to copy the database across, but I think that when we implemented the JUSP sync we stopped copying because of the overhead of copying the larger amount of data. I was meant to propose something to copy only the relevant parts of the db but I never did this - so ball in my court.

I guess this lack of copying may explain the difference?

Owen

Owen Stephens
Owen Stephens Consulting
Web: http://www.ostephens.com
Email: owen@ostephens.com
Telephone: 0121 288 6936
